File: ALGS200x/Week2/Challenge2-1/CPP/fib.cpp

#include <iostream>
#include <cassert>

int fib(int n)
{
  if (n <= 1)
  {
    return n;
  }
  long long a, b, count, tmp;
  count = 0;
  a = 0;
  b = 1;
  while (count++ < n)
  {
    tmp = a;
    a = b;
    b = tmp + a;
  }
  return a;
}

void run_tests()
{
  assert(fib(0) == 0);
  assert(fib(1) == 1);
  assert(fib(3) == 2);
  assert(fib(5) == 5);
  assert(fib(10) == 55);
  std::cout << "All tests pass." << std::endl;
  return;
}

int main()
{
  //run_tests();
  long long n;
  while (std::cin >> n)
  {
    std::cout << fib(n) << std::endl;
  }
}

File: ALGS200x/Week2/Challenge2-2/PY/fib.py
# python3

import sys

def fib(n):
    a, b = 0, 1
    count = 0
    while count < n:
        count = count + 1
        a, b = b, a + b
        a = a % 10
        b = b % 10
    return a

def run_tests():
    assert fib(0) == 0
    assert fib(1) == 1
    assert fib(2) == 1
    assert fib(3) == 2
    assert fib(5) == 5
    assert fib(7) == 3
    print("All tests pass.")

#run_tests()

if __name__ == "__main__":
    while True:
        try:
            n = int(input())
            print(fib(n))
        except EOFError:
            break
